src/compiler/checker.ts

```diff
@@ -1526,17 +1534,53 @@ module ts {
                     }
                 }
 
+                function writeSymbolTypeReference(symbol: Symbol, typeArguments: Type[], pos: number, end: number) {
+                    if (!isReservedMemberName(symbol.name)) {
+                        buildSymbolDisplay(symbol, writer, enclosingDeclaration, SymbolFlags.Type);

```

Comment:
If anything it should be `SymbolFlags.Type` or `SymbolFlags.Value`, but I'd rather leave it the way it is. It has to do with how aliases are resolved, but I don't think it matters. The sooner we can clean up that cryptic stuff the better.